api/kubevirt.io/resource/pci/v1alpha1/deviceclassparameters.go

```diff
@@ -0,0 +1,41 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeviceSelector allows one to match on a specific type of Device as part of the class.
+type DeviceSelector struct {
+	Type              string `json:"type"`
+	ResourceName      string `json:"resourceName"`
```

Comment:
Is it the `ResourceName` required? I think this could be replaced by the claim name

Reply:
permitted host devices in kubevirt CR uses the same name
```
configuration:
  permittedHostDevices:
    pciHostDevices:
    - pciVendorSelector: 8086:5845
      resourceName: devices.kubevirt.io/nvme
```
For this the resource class would look like this

```
apiVersion: pci.resource.kubevirt.io/v1alpha1
kind: DeviceClassParameters
metadata:
  name: pci-params
spec:
  deviceSelector:
    - resourceName: "devices.kubevirt.io/nvme"
      pciVendorSelector: "1b36:0010"
      type: "pci"
```
So I think resourceName should be consistent

Reply:
I think, we will need to modify kubevirt CR in order to list the allowed device classes/drivers instead of the resource names, but we can leave as it it for now.